packages/ai/core/embed/embed-many.test.ts

```diff
@@ -123,6 +123,28 @@ describe('options.headers', () => {
   });
 });
 
+describe('options.providerOptions', () => {
+  it('should pass provider options to model', async () => {
+    const result = await embedMany({
+      model: new MockEmbeddingModelV2({
+        doEmbed: async ({ providerOptions }) => {
+          expect(providerOptions).toStrictEqual({
+            aProvider: { someKey: 'someValue' },
+          });
+
+          return { embeddings: [[1, 2, 3]] };
+        },
+      }),
+      values: ['test-input'],
+      providerOptions: {
+        aProvider: { someKey: 'someValue' },
+      },
+    });
+
+    expect(result.embeddings).toStrictEqual([[1, 2, 3]]);
```

Comment:
how good are the error messages that we get when the provider options are not passed through? also the final expect is misleading.

I wonder if a `spy` pattern would be a better alternative for these kinds of tests, where we use a spy for `doEmbed` and then expect that the value sent into the spy is correct.